cli/spec.py

```diff
@@ -0,0 +1,244 @@
+            subcommand_options[subparser_name], _ = __process_options(
+                params, lookup=True, trim=True)
+
+    return global_options, subcommand_options
+
+
+def __get_groups_options(options, spec_dict, lookup, trim):
+    """
+    Gets the dict of options nested within the spec groups
+    """
+    if 'groups' in spec_dict:
+        for group in spec_dict['groups']:
+            __get_flat_options(options, group, lookup, trim)
+
+
+def __get_flat_options(options, spec_dict, lookup, trim):
+    """
+    Gest the dict of options listed in the spec of group.
+
+    This method will also remove some methods and will replace
+     __<value>__ pattens in the option properties (e.g. __default__, __FILE__)
+
+    """
+    if 'options' in spec_dict:
+        for opt_name, opt_params in spec_dict['options'].iteritems():
+            if lookup:
+                __lookup_option(opt_name, opt_params)
+
+            # get a parameters copy with all the keys.
+            options[opt_name] = dict(opt_params)
+
+            if trim:
+                __trim_option(opt_name, opt_params)
+
+
+def __lookup_option(opt_name, opt_params):
+    """
+    Modifies existing option parameters by replaceing ___*___ patterns
+      and inserting default values into the help.
+    :param opt_name: the option name to process
+    :param opt_params: the dictionary of the option parameters
+        (help, type, default, etc)
+    """
+
+    # first check __*__ pattern
+    for opt_param_name, param_value in opt_params.iteritems():
+        for lookup_string, lookup_key in LOOKUP_DICT.iteritems():
+            if lookup_string in str(param_value):
+                opt_params[opt_param_name] = \
+                    param_value.replace(lookup_string, str(opt_params.get(
+                        lookup_key, lookup_string)))
+
+    # insert default value into help.
+    if 'help' in opt_params and 'default' in opt_params:
+        opt_params['help'] += " | Default value: {}".format(
+            opt_params['default'])
+
+
+def __trim_option(opt_name, opt_params):
+    """
+    Removes the defined option parameters.
+    :param opt_name: the option name to process
+    :param opt_params: the dictionary of the option parameters
+        (help, type, default, etc)
+    """
+    for opt_param_name, param_value in opt_params.iteritems():
+        if opt_param_name in TRIM_PARAMS:
+            opt_params.pop(opt_param_name, None)
+
+
+def _get_specs(module_name, config):
+    """
+    Gets specs files as a dict from settings/<module_name> folder.
+    :param module_name: the module name: installer|provisioner|tester
+    """
+    res = {}
+    for spec_file in _get_all_specs(config, subfolder=module_name):
+        spec = yaml.load(
+            open(spec_file),
+            Loader=yamlordereddictloader.Loader)
+        utils.dict_merge(res, spec)
+    return res
+
+
+def _get_all_specs(config, subfolder=None):
```

Comment:
You can make this method more atomic by having only one parameter 'root_dir'. It'll be the method caller responsibility to pass the right path. In the method itself, you'll just have to verify that the path exists.